### DISCUSSION TITLE
How can I interpret the trace, or is there a documentation for the trace?

### DISCUSSION THREAD

**Discussion by sumin963** [UPPAALModelChecker/UPPAAL-Meta#148] [Q&A] [answered]
hello

I want to get some results through reinforcement learning of UPPAAL's trace.
However, when I opened the trace file, the values were as follows:
0
0
0
0
0
0
2
.
0
0
...
How can I interpret the trace, or is there a documentation for the trace?

thank you

**Comment by mikucionisaau** [comment, marked as answer]
There are two possibilities:
1. Connect to the engine through Java API, get that trace in a form of Java object and then examine it.
see `demo/ModelDemo.java` which uses `model.jar` (see also `model-javadoc.jar`, e.g. [javadoc](https://people.cs.aau.dk/~marius/modeldoc/)).
2. Use `tracer` from [UTAP](https://github.com/UPPAALModelChecker/utap) (C++ API) to load and interpret the trace, see [traces](https://github.com/UPPAALModelChecker/utap/wiki#traces).
